tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1278,6 +1292,85 @@ class CudnnRnnSequenceTensorDescriptor
   SE_DISALLOW_COPY_AND_ASSIGN(CudnnRnnSequenceTensorDescriptor);
 };
 
+class CudnnRnnVariableSequenceTensorDescriptor
```

Comment:
Let's not create a separate descriptor, instead, augment the existing `cudnnRnnSequenceTensorDescriptor`.
It should have the following new fields:
```py
RNNDataDescriptor
seq_lens
```
Rename the existing `seq_length_` to `max_seq_length_`.
The `CudnnRnnSequenceTensorDescriptor()` method would accept a new pointer `to seq_lengths`, which is set to `nullptr` if not using the new `xxxEx` APIs.